### PR TITLE
fix(core): workaround for broken file locking on Docker Desktop on MacOS

### PR DESCRIPTION
**Pull request by jerrinot**
See: https://github.com/docker/for-mac/issues/7004
and a minimized reproducer: https://github.com/jerrinot/flocktest

### CODE REVIEW

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 1]
```diff
@@ -906,14 +906,19 @@ public static boolean isValidTableName(CharSequence tableName, int fsFileNameLim
...
     public static int lock(FilesFacade ff, Path path, boolean verbose) {
-        final int fd = ff.openRW(path, CairoConfiguration.O_NONE);
-        if (fd == -1) {
-            if (verbose) {
-                LOG.error().$("cannot open '").utf8(path).$("' to lock [errno=").$(ff.errno()).I$();
+
+        // workaround for https://github.com/docker/for-mac/issues/7004
+        if (Files.VIRTIO_FS_DETECTED) {
+            if (!ff.touch(path)) {
+                LOG.error().$("cannot touch '").utf8(path).$("' to lock [errno=").$(ff.errno()).I$();
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/cairo/TableUtils.java:913]: Before this patch, all logging was made under the `verbose` flag check. Should we keep it?

Comment (jerrinot) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:913]: good catch, thanks!

Comment (jerrinot) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:913]: I cherry-picked the patch from my troubleshooting branch which was always logging (+a bunch more). 
but in prod code it should be guarded by `verbose`.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 12 (58.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|6|33.33%|
|:large_blue_circle:|io/questdb/std/Files.java|5|6|83.33%|

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 12 (58.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|6|33.33%|
|:large_blue_circle:|io/questdb/std/Files.java|5|6|83.33%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**